File: attendance-system-v2/backend/src/app/models/attendance.py
from sqlalchemy import Column, Integer, DateTime, ForeignKey, String
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.session import Base

class Attendance(Base):
    __tablename__ = "attendances"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"))
    check_in = Column(DateTime(timezone=True), server_default=func.now())
    check_out = Column(DateTime(timezone=True), nullable=True)
    status = Column(String, default="present") # present, late, absent
    
    user = relationship("User", back_populates="attendances")

File: attendance-system-v2/mobile_app/lib/models/attendance.dart
class Attendance {
  final int? id;
  final int userId;
  final DateTime checkIn;
  final DateTime? checkOut;
  final String status;
  final bool isSynced;

  Attendance({
    this.id,
    required this.userId,
    required this.checkIn,
    this.checkOut,
    required this.status,
    this.isSynced = true,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'],
      userId: json['user_id'],
      checkIn: DateTime.parse(json['check_in']),
      checkOut: json['check_out'] != null ? DateTime.parse(json['check_out']) : null,
      status: json['status'],
      isSynced: json['is_synced'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'check_in': checkIn.toIso8601String(),
      'check_out': checkOut?.toIso8601String(),
      'status': status,
      'is_synced': isSynced,
    };
  }
}
